HL/C compilation fails with Visual Studio when Haxe field is named restrict, _restrict, or __restrict

The C code for HashLink, when generated from the following Haxe code, cannot be compiled with Visual Studio's C compiler, cl.exe. It works with MinGW's gcc (and gcc on macOS and Linux), though.

```haxe
class HlcRestrict {
	static function main() {
		new HlcRestrict();
	}

	public function new() {
	}

	public var restrict:String;
}
```

Compiled with the following options:

```hxml
--main HlcRestrict
--hl generated/HlcRestrict.c
```

cl.exe command in Visual Studio developer shell:

```sh
 cl.exe /Ox /Fe:HlcRestrict.exe -I "<hashlink>\include" -I "generated" "generated/HlcRestrict.c" "<hashlink>\libhl.lib"
```

It results in the following error:

> generated\_std/HlcRestrict.h(22): error C2208: '_String *restrict ': no members defined using this type

Same result if you change it to `_restrict` or `__restrict` (one or two underscores).

```haxe
public var _restrict:String;
```

```haxe
public var __restrict:String;
```

However, if you change it to `___restrict` with three underscores, cl.exe can successfully compile the generated C code:

```haxe
public var ___restrict:String;
```

It looks like _src/generators/hl2c.ml_ already has some code to rewrite certain field names when they conflict with C keywords, including `restrict` actually, but I guess that this behavior is specific to Visual Studio's C compiler that appears to have been missed.

--- a/src/generators/hl2c.ml
+++ b/src/generators/hl2c.ml
@@ -98,13 +98,13 @@ let keywords =
 	"typeof";
 	(* C11 *)
 	"_Alignas";"_Alignof";"_Atomic";"_Bool";"_Complex";"_Generic";"_Imaginary";"_Noreturn";"_Static_assert";"_Thread_local";"_Pragma";
-	"inline";"restrict"
+	"inline";"restrict";"_restrict"
 	] in
 	let h = Hashtbl.create 0 in
 	List.iter (fun i -> Hashtbl.add h i ()) c_kwds;
 	h
 
-let ident i = if (Hashtbl.mem keywords i) || (ExtString.String.starts_with "__" i) then "_hx_" ^ i else i
+let ident i = if (Hashtbl.mem keywords i) || (ExtString.String.starts_with i "__") then "_hx_" ^ i else i
 
 let s_comp = function
 	| CLt -> "<"

--- a/tests/misc/hl/reservedKeywords/Macro.macro.hx
+++ b/tests/misc/hl/reservedKeywords/Macro.macro.hx
@@ -19,18 +19,26 @@ class Macro {
 			"typeof",
 			// C11
 			"_Alignas", "_Alignof", "_Atomic", "_Bool", "_Complex", "_Generic", "_Imaginary", "_Noreturn", "_Static_assert", "_Thread_local", "_Pragma",
-			"inline", "restrict"
+			"inline", "restrict", "_restrict"
 		];
 
 		var pos = Context.currentPos();
 
-		for (k in keywords)
+		for (k in keywords) {
 			fields.push({
 				pos: pos,
 				name: "_test_" + k,
 				meta: [{pos: pos, name: ":native", params: [macro $v{k}]}],
 				kind: FVar(macro :String, null)
 			});
+
+			fields.push({
+				pos: pos,
+				name: "_test2_" + k,
+				meta: [{pos: pos, name: ":native", params: [macro $v{'__' + k}]}],
+				kind: FVar(macro :String, null)
+			});
+		}
 
 		return fields;
 	}

